core/src/main/scala/org/apache/spark/memory/StaticMemoryManager.scala

```diff
@@ -72,8 +72,11 @@ private[spark] class StaticMemoryManager(
       numBytes: Long,
       evictedBlocks: mutable.Buffer[(BlockId, BlockStatus)]): Boolean = synchronized {
     val currentUnrollMemory = storageMemoryPool.memoryStore.currentUnrollMemory
-    val maxNumBytesToFree = math.max(0, maxMemoryToEvictForUnroll - currentUnrollMemory)
-    val numBytesToFree = math.min(numBytes, maxNumBytesToFree)
+    val freeMemory = storageMemoryPool.memoryFree
+    val maxNumBytesToFree = math.max(0,
+      maxMemoryToEvictForUnroll - currentUnrollMemory - freeMemory)
```

Comment:
In order to most closely match the 1.5.x behavior, shouldn't this be `maxMemoryToEvictForUnroll` (which should probably be renamed to `maxUnrollMemory` ?) minus `currentUnrollMemory`? See https://github.com/apache/spark/blame/branch-1.5/core/src/main/scala/org/apache/spark/storage/MemoryStore.scala#L290, where we don't take `freeMemory` into account when trying to ensure space.